Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Management/Merchant.hs

```diff
@@ -1250,7 +1261,7 @@ postMerchantConfigFarePolicyUpsert merchantShortId opCity req = do
           (oldFareProducts, newBoundedAlreadyDeletedMap) <-
             case timeBounds of
               Unbounded -> do
-                fareProducts <- maybeToList <$> CQFProduct.findUnboundedByMerchantVariantArea merchanOperatingCityId [searchSource] tripCategory vehicleServiceTier area
+                fareProducts <- SQF.findAllUnboundedByMerchantOpCityIdVariantArea merchanOperatingCityId area tripCategory vehicleServiceTier TB.Unbounded enabled' [searchSource]
```

Comment:
@blaze-198 Here, we should always delete the `enabled = True` fare product because that's what we are trying to disable.